Makefile

```diff
@@ -0,0 +1,146 @@
+
+## Create a docker image on disk for a specific arch and tag
+dist/flanneld-$(ARCH)-$(TAG).docker: dist/flanneld-$(ARCH)
+	docker build -f Dockerfile.$(ARCH) -t $(REGISTRY)/flannel-$(ARCH):$(TAG) .
+	docker save -o dist/flanneld-$(ARCH)-$(TAG).docker $(REGISTRY)/flannel-$(ARCH):$(TAG)
+
+# amd64 gets an image with the suffix too (i.e. it's the default)
+ifeq ($(ARCH),amd64)
+	docker build -f Dockerfile.$(ARCH) -t $(REGISTRY)/flannel:$(TAG) .
+endif
+
+## Create an ACI on disk for a specific arch and tag
+dist/flanneld-$(ARCH)-$(TAG).aci: dist/flanneld-$(ARCH)-$(TAG).docker
+	docker2aci dist/flanneld-$(ARCH)-$(TAG).docker
+	mv quay.io-coreos-flannel-$(ARCH)-$(TAG).aci dist/flanneld-$(ARCH)-$(TAG).aci
+	actool patch-manifest --replace --capability=CAP_NET_ADMIN \
+      --mounts=run-flannel,path=/run/flannel,readOnly=false:etc-ssl-etcd,path=/etc/ssl/etcd,readOnly=true:dev-net,path=/dev/net,readOnly=false \
+      dist/flanneld-$(ARCH)-$(TAG).aci
+
+docker-push: dist/flanneld-$(ARCH)-$(TAG).docker
+	docker push $(REGISTRY)/flannel-$(ARCH):$(TAG)
+
+# amd64 gets an image with the suffix too (i.e. it's the default)
+ifeq ($(ARCH),amd64)
+	docker push $(REGISTRY)/flannel:$(TAG)
+endif
+
+## Build an architecture specific static flanneld binary
+dist/flanneld-$(ARCH):
+    # Build for other platforms with ARCH=$ARCH make build
+    # valid values for $ARCH are [amd64 arm arm64 ppc64le]
+	docker run -e CC=$(CC) -e GOARM=$(GOARM) -e GOARCH=$(ARCH) -it \
+		-u $(shell id -u):$(shell id -u) \
+	    -v ${PWD}:/go/src/github.com/coreos/flannel:ro \
+        -v ${PWD}/dist:/go/src/github.com/coreos/flannel/dist \
+	    gcr.io/google_containers/kube-cross:$(KUBE_CROSS_TAG) /bin/bash -c '\
+		cd /go/src/github.com/coreos/flannel && \
+		CGO_ENABLED=1 make -e dist/flanneld && \
+		mv dist/flanneld dist/flanneld-$(ARCH) && \
+		file dist/flanneld-$(ARCH)'
+
+## Build an architecture specific iptables binary
+dist/iptables-$(ARCH):
+	docker run -e CC=$(CC) -e GOARM=$(GOARM) -e GOARCH=$(ARCH) -it \
+			-u $(shell id -u):$(shell id -u) \
+            -v ${PWD}:/go/src/github.com/coreos/flannel:ro \
+            -v ${PWD}/dist:/go/src/github.com/coreos/flannel/dist \
+            gcr.io/google_containers/kube-cross:$(KUBE_CROSS_TAG) /bin/bash -c '\
+            curl -sSL http://www.netfilter.org/projects/iptables/files/iptables-$(IPTABLES_VERSION).tar.bz2 | tar -jxv && \
+            cd iptables-$(IPTABLES_VERSION) && \
+            ./configure \
+                --prefix=/usr \
+                --mandir=/usr/man \
+                --disable-shared \
+                --disable-devel \
+                --disable-nftables \
+                --enable-static \
+                --host=amd64 && \
+            make && \
+            cp iptables/xtables-multi /go/src/github.com/coreos/flannel/dist/iptables-$(ARCH) && \
+            cd /go/src/github.com/coreos/flannel && \
+            file dist/iptables-$(ARCH)'
+
+## Build a .tar.gz for the amd64 flanneld binary
```

Comment:
IMHO container images that include the binary + dependencies are the way to go. No other binaries or archives needed. Feel free to ignore my radical opinion on this one